pkg/geoipupdate/database/local_file_writer.go

```diff
@@ -31,21 +27,21 @@ type LocalFileDatabaseWriter struct {
 // NewLocalFileDatabaseWriter create a LocalFileDatabaseWriter. It creates the
 // necessary lock and temporary files to protect the database from concurrent
 // writes.
-func NewLocalFileDatabaseWriter(filePath, lockFilePath string, verbose bool) (*LocalFileDatabaseWriter, error) {
+func NewLocalFileDatabaseWriter(filePath string, lock *FileLock, verbose bool) (*LocalFileDatabaseWriter, error) {
 	dbWriter := &LocalFileDatabaseWriter{
-		filePath:     filePath,
-		lockFilePath: lockFilePath,
-		verbose:      verbose,
+		filePath: filePath,
+		verbose:  verbose,
 	}
 
-	var err error
-	if dbWriter.lock, err = CreateLockFile(lockFilePath, verbose); err != nil {
+	if _, err := lock.acquireLock(); err != nil {
```

Comment:
Given that we are breaking the API anyway, I wonder if it would be better to just acquire the lock before downloading anything and then releasing it when we are all done rather than trying to acquire it  for each file.

If we did that, we could move the lock stuff into an `internal/` package.